@claude CI failed on HEAD `c4516e7d`. Please fix the failures below and push to this branch.

## Failed checks
- **github-actions** / `test-core` — https://github.com/MervinPraison/PraisonAI/actions/runs/31871927002/job/94982290337
- **github-actions** / `test-core (subdirs)` — https://github.com/MervinPraison/PraisonAI/actions/runs/31871927002/job/94981903385

## Failures (extracted)
1. `tests/unit/suite_runner/test_report_viewer.py::TestFindLatestReportDir::test_finds_latest_by_mtime` — `AssertionError: assert PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw0/test_finds_latest_by_mtime0/20260101_120000') == PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw0/test_finds_latest_by_mtime0/20260109_140000')`
   - Job: `test-core (subdirs)`

## Critical review first
Before changing code or tests, decide **which side is wrong**:
1. **Legitimate feature change** — the PR intent is correct but implementation or tests need updating. Preserve SDK guarantees; update tests only when behaviour intentionally changed and document why.
2. **Regression / bug in this PR** — the failure exposes a real breakage introduced here. Fix the implementation; **do not weaken, skip, or delete tests** just to go green.
3. **Pre-existing flake or unrelated failure** — say so explicitly; prefer fixing the root cause over masking it.

**SDK guardrails (AGENTS.md):**
- Agent-callable tools → PraisonAI-Tools; lifecycle plugins → PraisonAI-Plugins; sandbox backends → PraisonAI-Plugins (`praisonai.sandbox`) — not praisonaiagents/.
- Do not disturb core SDK contracts to accommodate a wrapper/feature change.
- Tests must continue to guard backward compatibility and hot-path behaviour — passing CI by lowering test standards is not acceptable.
- If the feature does not genuinely add SDK value, recommend reverting or narrowing scope instead of patching around failures.

## What to do
1. State your verdict: **legitimate fix**, **regression fix**, or **needs human review** — and why (1–3 sentences).
2. Fix root cause with **minimal changes**; never bloat the Agent class with extra params.
3. Run failing tests locally, e.g. `pytest tests/unit/suite_runner/test_report_viewer.py::TestFindLatestReportDir::test_finds_latest_by_mtime -q`, plus any related SDK tests touched by the PR.
4. Push to this branch and comment: files changed, review verdict, and why tests still protect SDK behaviour.

--- a/src/praisonai/praisonai/cli/commands/managed.py
+++ b/src/praisonai/praisonai/cli/commands/managed.py
@@ -663,7 +663,13 @@ _PS_PROVIDERS = ("docker", "e2b", "modal", "daytona", "flyio", "tenki")
 
 
 def _discover_instances(provider_filter: Optional[str] = None):
-    """Return [(provider_name, InstanceInfo), ...] for every reachable provider.
+    """Return (instances, errors) for reachable providers.
+
+    ``instances`` is [(provider_name, InstanceInfo), ...].
+    ``errors`` is [(provider_name, message), ...] for providers that were
+    *available* (installed + configured) but failed while listing — those must
+    surface, otherwise ``ps``/``stop --all`` would wrongly report an empty,
+    successful inventory while sandboxes keep running.
 
     Providers that are not installed or not configured are skipped silently --
     absence of an E2B key is a normal state, not an error worth printing.
@@ -674,26 +680,32 @@ def _discover_instances(provider_filter: Optional[str] = None):
 
     names = [provider_filter] if provider_filter else list(_PS_PROVIDERS)
     found = []
+    errors = []
 
     async def _collect():
         for name in names:
             try:
                 provider = resolve_compute(name)
             except Exception:
-                continue  # provider package not installed
+                # Not installed / unknown provider. Silent unless explicitly
+                # requested, where the user deserves to know it went nowhere.
+                if provider_filter:
+                    errors.append((name, "provider not installed or unknown"))
+                continue
             try:
                 if not getattr(provider, "is_available", True):
-                    continue
+                    continue  # daemon down / no credentials — a normal state
             except Exception:
-                continue  # daemon down / no credentials
+                continue
             try:
                 for info in await provider.list_instances():
                     found.append((name, info))
-            except Exception:
-                continue
+            except Exception as e:
+                # Available but listing failed: real error, never hide it.
+                errors.append((name, str(e)))
 
     asyncio.run(_collect())
-    return found
+    return found, errors
 
 
 def _format_uptime(created_at: float) -> str:
@@ -729,36 +741,44 @@ def managed_ps(
         praisonai managed ps --provider docker
         praisonai managed ps --json
     """
-    rows = _discover_instances(provider)
+    rows, errors = _discover_instances(provider)
 
     if as_json:
-        typer.echo(json.dumps([
-            {
-                "provider": name,
-                "instance_id": info.instance_id,
-                "status": getattr(info.status, "value", str(info.status)),
-                "endpoint": info.endpoint,
-                "created_at": info.created_at,
-                "metadata": info.metadata,
-            }
-            for name, info in rows
-        ], indent=2))
-        return
+        typer.echo(json.dumps({
+            "sandboxes": [
+                {
+                    "provider": name,
+                    "instance_id": info.instance_id,
+                    "status": getattr(info.status, "value", str(info.status)),
+                    "endpoint": info.endpoint,
+                    "created_at": info.created_at,
+                    "metadata": info.metadata,
+                }
+                for name, info in rows
+            ],
+            "errors": [{"provider": name, "error": msg} for name, msg in errors],
+        }, indent=2))
+        raise typer.Exit(1 if errors else 0)
 
     if not rows:
         typer.echo("No running sandboxes.")
-        return
+    else:
+        typer.echo(f"{'INSTANCE ID':<26} {'PROVIDER':<10} {'STATUS':<10} {'UPTIME':<8} IMAGE")
+        typer.echo("-" * 78)
+        for name, info in rows:
+            status = getattr(info.status, "value", str(info.status))
+            image = (info.metadata or {}).get("image") or info.endpoint or ""
+            typer.echo(
+                f"{info.instance_id:<26} {name:<10} {status:<10} "
+                f"{_format_uptime(info.created_at):<8} {image}"
+            )
+        typer.echo(f"\n{len(rows)} running. Stop with: praisonai managed stop <instance-id>")
 
-    typer.echo(f"{'INSTANCE ID':<26} {'PROVIDER':<10} {'STATUS':<10} {'UPTIME':<8} IMAGE")
-    typer.echo("-" * 78)
-    for name, info in rows:
-        status = getattr(info.status, "value", str(info.status))
-        image = (info.metadata or {}).get("image") or info.endpoint or ""
-        typer.echo(
-            f"{info.instance_id:<26} {name:<10} {status:<10} "
-            f"{_format_uptime(info.created_at):<8} {image}"
-        )
-    typer.echo(f"\n{len(rows)} running. Stop with: praisonai managed stop <instance-id>")
+    if errors:
+        typer.echo("\nSome providers could not be queried (sandboxes may still be running):")
+        for name, msg in errors:
+            typer.echo(f"  {name}: {msg}")
+        raise typer.Exit(1)
 
 
 @app.command("stop")
@@ -784,14 +804,14 @@ def managed_stop(
         typer.echo("Give an instance ID or --all. See: praisonai managed ps")
         raise typer.Exit(1)
 
-    rows = _discover_instances(provider)
+    rows, errors = _discover_instances(provider)
     if instance_id:
         rows = [r for r in rows if r[1].instance_id == instance_id]
         if not rows:
             typer.echo(f"No running sandbox with ID {instance_id!r}. See: praisonai managed ps")
             raise typer.Exit(1)
 
-    if not rows:
+    if not rows and not errors:
         typer.echo("No running sandboxes.")
         return
 
@@ -806,7 +826,15 @@ def managed_stop(
             failed += 1
 
     typer.echo(f"\n{stopped} stopped" + (f", {failed} failed" if failed else ""))
-    if failed:
+
+    if errors:
+        # A provider we could not even inventory may still hold live sandboxes
+        # we never got the chance to stop. Never claim a clean sweep.
+        typer.echo("\nSome providers could not be queried (sandboxes may still be running):")
+        for name, msg in errors:
+            typer.echo(f"  {name}: {msg}")
+
+    if failed or errors:
         raise typer.Exit(1)
 
 

--- a/src/praisonai/praisonai/integrations/managed_agents.py
+++ b/src/praisonai/praisonai/integrations/managed_agents.py
@@ -1092,10 +1092,10 @@ def ManagedAgent(
       (It does NOT raise; prefer LocalAgent(compute="...") directly.)
     
     Returns:
-        An instance satisfying ``ManagedBackendProtocol``.
-        
-    Raises:
-        ValueError: For compute-provider names that should use LocalAgent(compute=).
+        An instance satisfying ``ManagedBackendProtocol``. Compute-provider
+        names ({"e2b","modal","flyio","daytona","docker","tenki"}) do NOT
+        raise — they emit a ``DeprecationWarning`` and delegate to
+        ``LocalManagedAgent(compute=<provider>)``.
     """
     # Track if provider was auto-detected to avoid spurious deprecation warnings
     auto_detected = provider is None

--- a/src/praisonai/praisonai/suite_runner/report_viewer.py
+++ b/src/praisonai/praisonai/suite_runner/report_viewer.py
@@ -190,8 +190,9 @@ def find_latest_report_dir(suite: str, base_dir: Optional[Path] = None) -> Optio
     if not candidates:
         return None
     
-    # Sort by mtime (newest first)
-    candidates.sort(key=lambda p: p.stat().st_mtime, reverse=True)
+    # Sort by mtime (newest first), breaking ties by name so timestamp-named
+    # dirs (e.g. "20260109_140000") remain deterministic when mtimes collide.
+    candidates.sort(key=lambda p: (p.stat().st_mtime, p.name), reverse=True)
     return candidates[0]
 
 

--- a/src/praisonai/tests/unit/test_managed_ps.py
+++ b/src/praisonai/tests/unit/test_managed_ps.py
@@ -97,13 +97,15 @@ def test_ps_json_is_machine_readable(runner, one_docker_instance):
     result = runner.invoke(app, ["ps", "--json"])
     assert result.exit_code == 0
     payload = json.loads(result.output)
-    assert payload[0]["instance_id"] == "docker_abc123"
-    assert payload[0]["provider"] == "docker"
-    assert payload[0]["status"] == "running"
+    sandboxes = payload["sandboxes"]
+    assert sandboxes[0]["instance_id"] == "docker_abc123"
+    assert sandboxes[0]["provider"] == "docker"
+    assert sandboxes[0]["status"] == "running"
+    assert payload["errors"] == []
 
 
 def test_ps_reports_empty_state(runner, monkeypatch):
-    monkeypatch.setattr(managed_cli, "_discover_instances", lambda *a, **k: [])
+    monkeypatch.setattr(managed_cli, "_discover_instances", lambda *a, **k: ([], []))
     result = runner.invoke(app, ["ps"])
     assert result.exit_code == 0
     assert "No running sandboxes" in result.output
@@ -119,6 +121,35 @@ def test_ps_skips_unavailable_providers(runner, monkeypatch):
     result = runner.invoke(app, ["ps"])
     assert result.exit_code == 0
     assert "never_shown" not in result.output
+
+
+def test_ps_surfaces_listing_failure(runner, monkeypatch):
+    """An available provider that fails to list must NOT report clean success."""
+    class Broken(FakeProvider):
+        async def list_instances(self):
+            raise RuntimeError("daemon exploded")
+
+    monkeypatch.setattr(managed_cli, "_PS_PROVIDERS", ("docker",))
+    monkeypatch.setattr(
+        "praisonaiagents.managed._compute_bridge.resolve_compute",
+        lambda name: Broken(available=True),
+    )
+    result = runner.invoke(app, ["ps"])
+    assert result.exit_code == 1
+    assert "daemon exploded" in result.output
+
+
+def test_ps_explicit_unknown_provider_is_loud(runner, monkeypatch):
+    """`--provider foo` for an uninstalled/unknown provider should not be silent."""
+    def fake_resolve(name):
+        raise ImportError(f"{name} not installed")
+
+    monkeypatch.setattr(
+        "praisonaiagents.managed._compute_bridge.resolve_compute", fake_resolve
+    )
+    result = runner.invoke(app, ["ps", "--provider", "e2b"])
+    assert result.exit_code == 1
+    assert "e2b" in result.output
 
 
 # ── stop ─────────────────────────────────────────────────────────────────────

